src/Microsoft.DotNet.Watcher.Core/Internal/FileWatcher/PollingFileWatcher.cs

```diff
@@ -117,13 +117,20 @@ private void CheckForChangedFiles()
                 {
                     var fileMeta = _knownEntities[fullFilePath];
 
-                    if (fileMeta.FileInfo.LastWriteTime != f.LastWriteTime)
+                    try
                     {
-                        // File changed
-                        RecordChange(f);
-                    }
+                        if (fileMeta.FileInfo.LastWriteTime != f.LastWriteTime)
+                        {
+                            // File changed
+                            RecordChange(f);
+                        }
 
-                    _knownEntities[fullFilePath] = new FileMeta(fileMeta.FileInfo, true);
+                        _knownEntities[fullFilePath] = new FileMeta(fileMeta.FileInfo, true);
+                    }
+                    catch(FileNotFoundException)
```

Comment:
Hm... I think were's okay with just the try-catch because `ForeachEntityInDirectory` calls `dirInfo.EnumerateFileSystemInfos("*.*");` which returns files that exist at the time of the invocation.